src/executors/CExecutor.ts

```diff
@@ -0,0 +1,64 @@
+import NonInteractiveCodeExecutor from './NonInteractiveCodeExecutor';
+import * as child_process from "child_process";
+import type {ChildProcessWithoutNullStreams} from "child_process";
+import type {Outputter} from "src/Outputter";
+import type {ExecutorSettings} from "src/settings/Settings";
+
+export default class CExecutor extends NonInteractiveCodeExecutor {
+
+	constructor(settings: ExecutorSettings, file: string) {
+		super(settings, false, file, "c");
+	}
+
+	override run(codeBlockContent: string, outputter: Outputter, cmd: string, cmdArgs: string, ext: string) {
+		const extension = "cpp";
+		const args = `-std=${this.settings.clingStd} ${this.settings.clingArgs}`;
+		
+		codeBlockContent = `extern "C" {
+			${codeBlockContent}
+		}`
+
+		// Run code with a main block
+		if (this.settings.cppUseMain) {
```

Comment:
Should this be `this.settings.cUseMain`?

Reply:
Yes, thanks for catching that!